norminette/lexer/lexer.py

```diff
@@ -26,34 +24,59 @@ def __init__(self, file: File):
 
         self.src = file.source
         self.len = len(file.source)
-        self.__char = self.src[0] if self.src != "" else None
         self.__pos = int(0)
         self.__line_pos = self.__line = 1
         self.tokens = []
 
     def peek_sub_string(self, size):
         return self.src[self.__pos : self.__pos + size]
 
-    def peek(self, *, offset: int = 0, collect: int = 1):
+    def raw_peek(self, *, offset: int = 0, collect: int = 1):
         assert collect > 0 and offset >= 0
         if (pos := self.__pos + offset) < self.len:
             return ''.join(self.src[pos:pos+collect])
         return None
 
+    def peek(self, *, offset: int = 0) -> Optional[Tuple[str, int]]:
+        if (trigraph := self.raw_peek(offset=offset, collect=3)) in trigraphs:
+            return trigraphs[trigraph], 3
+        if (digraph := self.raw_peek(offset=offset, collect=2)) in digraphs:
+            return digraphs[digraph], 2
+        if word := self.raw_peek(offset=offset):
+            return word, 1
+        return None  # Let it crash :D
+
     def pop(self, *, times: int = 1, use_spaces: bool = False):
         assert times > 0
         result = ""
         for _ in range(times):
-            char = self.peek()
+            for _ in range(100):
+                char, size = self.peek()
+                if char != '\\':
+                    break
+                if self.peek(offset=size) is None:
+                    break
+                temp, _ = self.peek(offset=size)  # Don't change the `temp` to `char`
+                if temp != '\n':
+                    break
+                self.__pos += size + 1
+                self.__line += 1
+                self.__line_pos = 0
+                if self.peek() is None:
+                    raise UnexpectedEOF()
+                char, size = self.peek()
+                break
+            else:
+                raise MaybeInfiniteLoop()
```

Comment:
I don't really understand the `else`. When did this loop continue? for what I understand at first iteration worst case you hit le break line 68.

Reply:
The `break` on line 68 should not exist. I added more information in [18d9e80](https://github.com/42School/norminette/pull/486/commits/18d9e80b00ca4bff5f5f548589753925f15a1b5a).